tools/ci_connector_ops/ci_connector_ops/utils.py

```diff
@@ -2,21 +2,15 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
-
 from dataclasses import dataclass
 import logging
 from pathlib import Path
 from typing import Dict, Optional, Set, Tuple, List
-import os
 import git
 import requests
 import yaml
 
-# ensure we are at the repository root
-os.chdir(os.path.dirname(os.path.abspath(__file__)))
-os.chdir('../../..')
-
-AIRBYTE_REPO = git.Repo(".")
+AIRBYTE_REPO = git.Repo(search_parent_directories=True)
```

Comment:
Oh, of course - the magic option was set to False...
Would this still work if we added back the `os.chdir('../../..')`?

Reply:
This would not unfortunately.

The issue here came down to the file found it self in different directories and running contexts.

So change the directory via a relative path would always break one of `local` or `ec2`

On `local` the paths were:
- current working directory:  /Users/ben/Development/repos/airbyte/tools/ci_connector_ops
- file directory: /Users/ben/Development/repos/airbyte/tools/ci_connector_ops/ci_connector_ops

On the `ec2` instance the paths were (taken from a [workflow we ran](https://github.com/airbytehq/airbyte/actions/runs/4078521733/jobs/7028859208)):
- The current working directory was `/actions-runner/_work/airbyte/airbyte`
- And the files path was `/actions-runner/_work/_tool/Python/3.9.16/x64/lib/python3.9/site-packages/ci_connector_ops/utils.py`

So our best way seems to be relying on using the GitPython library to simply traverse up the directory structure until it finds a `.git` folder

Reply:
🤔 Thinking on that a bit. Perhaps if theres is a case where we would not be inside a git versioned directory.

@alafanechere Is there a context that might happen?